OpenMage/magento-lts
# 20.0

### Description (*)
<! ---
   Enhance OpenMage installation / update instructions.
->


### Issues fixed (if applicable)

1. Fix OpenMage / magento-lts # 1482

### Manual Test Scenarios (*)

1 - suppose you explain to a total inexperienced how to install OpenMage
2 - List the steps required for installation / upgrade

In the past updating Magento was easy, you used MAGENTO CONNECT and it was done.

If you have purchased or found a free module, download the package and upload it from MAGENTO CONNECT.
Anyone could do it.

the old MAGENTO can be installed almost anywhere, even on low cost hosting.

With the advent of COMPOSER, things have changed radically ..

Magento is no longer an e-commerce for everyone.

From the very first steps you are faced with enormous difficulties for those who are not familiar with the composer or even worse for those who have a very limited hosting that does not have ssh or installation permissions.

The impression is that the standard is aimed at preventing the access of the software to those who want to use it.

In magento 2 it is even worse, first they inserted a MarketPlace to easily install the modules and then removed ... (from the series we complicate life)

Is OpenMage going the same way?
some examples?
Arrival from Magento 1.9.4.5
I tried to install OpenMage via composer following the instructions in GitHub
I have not succeeded.
The instructions are monstrously incomplete and misleading.
I only managed to install it with the old classic method:

1 - Download the packages
2 - Unpack the package
3 - install via browser

Problem: Unable to update (with current instructions).

FURTHERMORE
you removed the magento compilation because you get excellent results with REDIS (honestly I tried to make a comparison between a magento 1.9.4.5 installation compiled on a shared server without redis and an OPENMAGE installation on a cloud server plus, optimized for magento2 with REDIS + google pagespeed etc etc and the result was that Openmage was slower. One wonders if deleting the compilation was a wise move, couldn't you leave it to the users' discretion?)

You have assumed that EVERYONE has REDIS available
anyway
SOLUTION:

Premise:
1 - you must have access to the server / hosting via ssh
2 - you must be able to use the composer
3 - If you have customized the RWD template without creating a custom template, make sure to backup the CSS and RWD folders before updating
4 - If you have installed OpenMage in the classic way via Browser go to point "5.1" (replace the name of the "openmage" folder with that of the folder in which your OpenMage installation is active)
5 - The OpenMage destination folder cannot be the main root of your web server / hosting example:
www / ..
or
yourvirtualhosting / www / ..

Let's begin:

1.1 - Download the compressed OpenMage-lts file from github
2.1 - Upload the compressed OpenMage-lts file to your server:
www / magento-lts-20-x.zip
or
yourvirtualhosting / www / magento-lts-20-x.zip

3.1 - Unzip the file
4.1 - Rename the generated folder with a simple name like "openamge" to get something like this:
www / openmage
or
yourvirtualhosting / www / openmage

5.1 - in the new "openmage" folder find and copy the composer.json file in the top level directory, you need to get this result:

www / composer.json
or
yourvirtualhosting / www / composer.json

6.1 - open to edit the "www / composer.json" file with a text editor or nano.

7.1 - find and edit the line like this:
from so
"magento-root-dir": "htdocs"
to so:
"magento-root-dir": "openmage"
8.1 - save

9.1 from the ssh terminal run composer making sure you are in:
# / path / directory / server / in / site / web / www
and not in
# / path / directory / server / in / site / web / www / openmage

10.1 - run the command:
composer update

At the end of the operation you will have updated openmage to the latest version.

this is my proposal.
if anyone wants to add comments, shell commands to change directories, unzip compressed files, copy files from one folder to another and edit them with nano, they are free to do so.
